errors.go

```diff
@@ -325,16 +325,16 @@ func decodeError(err error) error {
 	switch err := err.(type) {
 	case *url.Error:
 		if _, ok := err.Err.(net.Error); ok {
-			err := fmt.Errorf("we found a problem connected against the api")
+			err := fmt.Errorf("problem connecting to the API")
```

Comment:
we could return also the error to be more descriptive and allow the user to debug more precisely, what do you think?

this is valid for all the errors in the PR

Reply:
I think we need to implement a global flag `--debug` and if that is true we can show more verbose errors

Reply:
I'm not against having a flag or more descriptive errors overall, but just raised this because during a connection problem I noticed the language currently isn't especially clear.

Reply:
I'm ok with this one, we can put the flag for feature versions